packages/flutter_tools/lib/src/isolated/resident_web_runner.dart

```diff
@@ -608,9 +608,15 @@ Please provide a valid TCP port (an integer between 0 and 65535, inclusive).
   }) async {
     if (_chromiumLauncher != null) {
       final Chromium chrome = await _chromiumLauncher!.connectedInstance;
-      final ChromeTab? chromeTab = await chrome.chromeConnection.getTab((ChromeTab chromeTab) {
-        return !chromeTab.url.startsWith('chrome-extension');
-      }, retryFor: const Duration(seconds: 5));
+      late final ChromeTab? chromeTab;
+      try {
+        chromeTab = await chrome.chromeConnection.getTab((ChromeTab chromeTab) {
+          return !chromeTab.url.startsWith('chrome-extension');
+        }, retryFor: const Duration(seconds: 5));
+      } on IOException {
+        // We were unable to unable to communicate with Chrome.
+        chromeTab = null;
+      }
```

Comment:
This should already be caught by the recently added

https://github.com/flutter/flutter/blob/8b3cd9a8d4db6baa03bf1ae8326254fed2b4e942/packages/flutter_tools/lib/src/isolated/resident_web_runner.dart#L386-L389

However, I prefer trying to catch exceptions as close to the thrower as we can.

Reply:
The semantics are different whether this is caught at line 386 or 616. That is to say:

1. should we be calling `appFailedToStart()` here?
2. should we be printing to STDERR the error and stacktrace?
3. should we be calling the more explicit `HttpException` here?
4. on the other hand, should we be calling the more general `IOException` on line 386?

I'm genuinely asking, I don't know this code at all.